enterprise/frontend/src/metabase-enterprise/content_translation/components/ContentTranslationConfiguration.tsx

```diff
@@ -0,0 +1,158 @@
+import {
+  type ChangeEvent,
+  type ReactNode,
+  useCallback,
+  useRef,
+  useState,
+} from "react";
+import { t } from "ttag";
+
+import { UploadInput } from "metabase/components/upload";
+import ExternalLink from "metabase/core/components/ExternalLink";
+import Markdown from "metabase/core/components/Markdown";
+import {
+  Form,
+  FormProvider,
+  FormSubmitButton,
+  useFormContext,
+} from "metabase/forms";
+import { useSelector } from "metabase/lib/redux";
+import { getDocsUrl } from "metabase/selectors/settings";
+import { Group, Icon, Loader, Stack, Text } from "metabase/ui";
+import { useUploadContentTranslationDictionaryMutation } from "metabase-enterprise/api";
+
+export const ContentTranslationConfiguration = () => {
+  const availableLocalesDocsUrl = useSelector((state) =>
+    // eslint-disable-next-line no-unconditional-metabase-links-render -- This is used in admin settings
+    getDocsUrl(state, {
+      page: "configuring-metabase/localization",
+      anchor: "supported-languages",
+    }),
+  );
+
+  return (
+    <Stack gap="sm" maw="38rem">
+      <Text>
+        <Markdown
+          components={{
+            strong: ({ children }: { children: ReactNode }) => (
+              <ExternalLink href={availableLocalesDocsUrl}>
+                {children}
+              </ExternalLink>
+            ),
+          }}
+        >
+          {t`You can upload a translation dictionary. We'll use this to translate user-provided strings (like question names) into the viewer's language. (Built-in strings won't be affected.) Your translation dictionary should be a CSV with three columns: Locale code, String, Translation. Supported locale codes are **listed here**. Uploading a new dictionary will replace the existing translations.`}
+        </Markdown>
+      </Text>
+
+      <FormProvider
+        // We're only using Formik to make the appearance of the submit button
+        // depend on the form's status. We're not using Formik's other features
+        // here.
+        initialValues={{}}
+        onSubmit={() => {}}
+      >
+        <UploadForm />
+      </FormProvider>
+    </Stack>
+  );
+};
+
+const UploadForm = () => {
+  const [uploadContentTranslationDictionary] =
+    useUploadContentTranslationDictionaryMutation();
+
+  // This is used to reset the input so that the same file can be uploaded
+  // multiple times
+  const [uploadInputReactKey, setUploadInputReactKey] = useState(0);
```

Comment:
this works, but it's a little convoluted. check out `frontend/src/metabase/collections/components/CollectionHeader/CollectionUpload.tsx` which just clears the value after upload, which is, I think, a more straightforward approach than triggering a re-render via a state update.

Reply:
I agree - resetting the input directly is more conventional and clear